PR: https://github.com/odoo/odoo/pull/206494

From: 6422427ce10dfbe74186985bfe8da540d11cebde
From: igbe
Date: 2025-06-23 08:59:10

Breaking data model changes scores: del:2 + add:0, change matches:
-    _inherit = "product.category"
-    property_account_downpayment_categ_id = fields.Many2one(

Total Changes: 34

[IMP] sale, l10n_fr_account, l10n_be: improve down payment

Move the property_account_downpayment_categ_id field to the company (instead of product.category).

The reason for this is we want to simplify the downpayment accounts. Having one account per product category could lead to downpayment invoices with a lot of lines, as we split lines based on account. Now that we have only 1 account per company, we only split lines based on taxes.

task-4640319

Part-of: odoo/odoo#206494
Related: odoo/enterprise#88054
Related: odoo/upgrade#7580
Signed-off-by: John Laterre (jol) <jol@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale/models/__init__.py
+++ b/addons/sale/models/__init__.py
@@ -9,7 +9,6 @@ from . import ir_actions_report
 from . import ir_config_parameter
 from . import payment_provider
 from . import payment_transaction
-from . import product_category
 from . import product_document
 from . import product_pricelist_item
 from . import product_product

--- a/addons/sale/models/chart_template.py
+++ b/addons/sale/models/chart_template.py
@@ -6,5 +6,5 @@ class AccountChartTemplate(models.AbstractModel):
 
     def _get_property_accounts(self, additional_properties):
         property_accounts = super()._get_property_accounts(additional_properties)
-        property_accounts['property_account_downpayment_categ_id'] = 'product.category'
+        property_accounts['downpayment_account_id'] = 'res.company'
         return property_accounts

--- a/addons/sale/models/product_category.py
+++ b/None
@@ -1,16 +0,0 @@
-from odoo import fields, models
-
-
-class ProductCategory(models.Model):
-    _inherit = "product.category"
-
-    property_account_downpayment_categ_id = fields.Many2one(
-        comodel_name='account.account',
-        company_dependent=True,
-        string="Downpayment Account",
-        domain=[
-            ('account_type', 'not in', ('asset_receivable', 'liability_payable', 'asset_cash', 'liability_credit_card', 'off_balance'))
-        ],
-        help="This account will be used on Downpayment invoices.",
-        tracking=True,
-    )

--- a/addons/sale/models/product_template.py
+++ b/addons/sale/models/product_template.py
@@ -230,11 +230,6 @@ class ProductTemplate(models.Model):
         """
         return ['no']
 
-    def _get_product_accounts(self):
-        product_accounts = super()._get_product_accounts()
-        product_accounts['downpayment'] = self.categ_id.property_account_downpayment_categ_id
-        return product_accounts
-
     ####################################
     # Product/combo configurator hooks #
     ####################################

--- a/addons/sale/models/res_company.py
+++ b/addons/sale/models/res_company.py
@@ -47,6 +47,16 @@ class ResCompany(models.Model):
         ],
         string="Sale onboarding selected payment method")
 
+    downpayment_account_id = fields.Many2one(
+        comodel_name='account.account',
+        string="Downpayment Account",
+        domain=[
+            ('account_type', 'in', ('income', 'income_other', 'liability_current')),
+        ],
+        help="This account will be used on Downpayment invoices.",
+        tracking=True,
+    )
+
     @api.constrains('prepayment_percent')
     def _check_prepayment_percent(self):
         for company in self:
